scripts/evaluate_release_tag.py

```diff
@@ -63,6 +63,9 @@ def should_build_docs(tag: Version) -> bool:
     elif tag >= latest_version:
```

Comment:
Are these conditions used only to set relevant print statements? 
if these print statements are not needed then the code can be rephrased as below,
```need_to_build_docs = False
    need_to_build_docs = True
    if not is_plain_version(tag):
        print(f"Tag {tag} is an alpha, rc, nightly, or otherwise non-standard version.")
        need_to_build_docs = False
```

if the print statements are needed, then the assignment can't avoided for every condition like this,
```
    need_to_build_docs = True
    if not is_plain_version(tag):
        print(f"Tag {tag} is an alpha, rc, nightly, or otherwise non-standard version.")
        need_to_build_docs = False
    elif tag >= latest_version:
        print(f"Tag {tag} is the latest version. Docs should be built.")
    elif tag < latest_version:
        print(f"Tag {tag} is an old minor. Docs should be built.")
    elif tag.major == previous_major and tag > previous_major_latest_version:
        print(f"Tag {tag} is higher than the latest version for the previous major. Docs should be built.")
```

Reply:
@varunshankar Sorry for the delay had to confirm this behaviour with @radovanZRasa . This is not an issue, the docs publish should have failed. We just release docs for latest minor and since your fix's docs was already released with 3.4.4 the docs didnot need to be updated again in 3.3.4 (for the same fix). I am closing this PR.